### PR TITLE
Issue/expand collapse grouped products

### PR DESCRIPTION
**Pull request by atorresveiga**
### Description
This PR adds collapse/expand behavior to grouped products. It does that by adding the new property isExpanded to Grouped products and check for that value to represent the state of the item (expanded/collapsed)

### Testing instructions
1. Open the orders list
2. Select an order containing bundled products or composite products
3. Check that now grouped products can be expanded/collapsed
4. Expand a grouped product and rotate the screen
5. Check that the expanded product maintains its state (it remains expanded)

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/46d607f8-d890-420e-9510-7198fc7645a0

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt [hunk 1]
```diff
@@ -787,6 +788,14 @@ class OrderDetailViewModel @Inject constructor(
         shippingLabelOnboardingRepository.markWcShippingBannerAsDismissed()
     }
 
+    fun onIsExpandedChange(groupedProductItem: OrderProduct.GroupedProductItem) {
+        _productList.value?.firstOrNull { orderProduct ->
+            (orderProduct as? OrderProduct.GroupedProductItem)?.product == groupedProductItem.product
+        }?.let { orderProduct ->
+            (orderProduct as? OrderProduct.GroupedProductItem)?.isExpanded == groupedProductItem.isExpanded
+        }
+    }
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt]: Maybe I'm missing something, but is this method doing anything? Executing it, doesn't seem to have any effect (no return value, no state modification etc.)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt [hunk 2]
```diff
@@ -607,7 +608,7 @@ class OrderDetailViewModel @Inject constructor(
     ): ListInfo<OrderProduct> {
         val products = refunds.list.getNonRefundedProducts(order.items)
         checkAddonAvailability(products)
-        val orderProducts = products.toOrderProducts()
+        val orderProducts = orderProductMapper.toOrderProducts(_productList.value ?: emptyList(), products)
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt:611]: I was always concerned that sometimes we map data in ViewModel and on the main thread. It's especially dangerous, as our mapping involves JSON serializations.

It's not feedback for this PR (I understand it was like this previously), just a thought leaving for possible improvements in the future.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt (hunk 1 above)
Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt]: Good catch! I miss to remove it before sending the PR to review, sorry 🤦 
Fixed https://github.com/woocommerce/woocommerce-android/pull/9086/commits/74cd77f572011de33ed2bf2da6bc2ffaaaea1893

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt (hunk 2 above)
Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt:611]: >  just a thought leaving for possible improvements in the future.

I think that maybe we can take this into account as part of the performance improvement project. Maybe the UI will be less laggy using a different thread, like `computation` for these mapping tasks

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt (hunk 1 above)
Comment (wzieba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt]: oh okay, gotcha! Thanks for addressing it

**Review by wzieba — APPROVED** [review]
Looks good! 

https://github.com/woocommerce/woocommerce-android/assets/5845095/4e8bc61e-02c2-48b2-91bf-d85c67d02895

**Pull request merged**